.github/workflows/clang-format.yml

```diff
@@ -18,6 +18,11 @@ name: clang-format
 on:
   pull_request:
 
+# Unique to this workflow and PR, or SHA if not a PR
```

Comment:
For the linting ones where the run-time is measured in seconds rather than minutes I don't see any point in having this.  It's unlikely to ever trigger, it won't save much in the way of resources when it does and it's extra overhead to maintain.

Reply:
Another problem I've ran into is:
- I push some commit A
- GHA is really backed up and takes a while to start some jobs
- I push a new commit B
- GHA linting for commit A runs and reports failures, but it's already been addressed in commit B
- I get confused

Reply:
It might be seconds now, but if/when we add longer running checks it will become an issue. I'm trying to proactively trim the GHA before we run into the problem we have with Jenkins runs: waaaaay too much being ran waaay too often and stopping progress

Reply:
The longer running checks (or at least ci2) already has this feature enabled.  One thing I don't like with this however is that if you push commit A and then push commit B before GHA has completed which is easy for ci2 (it happens for example if you need to re-push for a linting error) then you get an email about the jobs for the push A being killed which is also confusing.

Jenkins is different as it's the pipeline that's killed, it's all or nothing so you have to kill it.  As I say for longer running jobs in GHA this is a no-brainer but linting jobs as a whole should be under the threshold where there's a benefit.

I think GHA rate-limiting is per person by the way, I've certainly had a couple of days where I've been iterating on GHA triggers and just stopped getting results for hours.

Reply:
> if you push commit A and then push commit B before GHA has completed which is easy for ci2 (it happens for example if you need to re-push for a linting error) then you get an email about the jobs for the push A being killed which is also confusing.

Yeah I agree. And unfortunately it's a known issue that hasn't been addressed: https://github.com/orgs/community/discussions/13015

Even if we don't **_need_** it for short jobs, I don't think the maintenance is too much since it's just copy paste across workflows. And I think it's good to not waste the resources. We can't keep adding more GHA without regard for resource usage. The sooner we start addressing that, the easier it is to manage in the future.

Reply:
We can't, hence my initial comment about short-running linting jobs only.

This will almost never trigger, and then it does it'll make the user experience worse but save minimal cycles.  The purpose of CI is to use compute resources to assist the developer, saving resources at the cost of developer overhead is not the right tradeoff in this case IMO.

Reply:
> it'll make the user experience worse but save minimal cycles

I disagree. Yes, it will almost never trigger. But if it does trigger, it's probably because the workflows are backed up. IMO, the experience is worse if I suddenly get a flood of failed lint checks.

Anyway, I'll remove it for the short jobs since it's not a perfect solution anyway. Mostly because GitHub sends a notification when the workflow is cancelled.

Reply:
Oh. This one runs in a few seconds but already had the `cancel-in-progress`. So we're already doing this for some short running jobs

Reply:
I guess it also gets confusing to wonder why some jobs get cancelled and some don't. It's just arbitrary at this point

Reply:
> Oh. This one runs in a few seconds but already had the `cancel-in-progress`. So we're already doing this for some short running jobs https://github.com/daos-stack/daos/actions/runs/8727893484/job/23946437185

is that the bash one? That runs on our network so won't be subject to scheduling delays and it takes less than 5 seconds.  There's next to zero chance that will ever actually trigger.

We need a threshold and yes, it's arbitrary but we have lots of jobs that run for less than a minute and a couple that run for more than 15 minutes, I'm not aware of any that take between 1 and 15 minutes.